ra_aid/agent_utils.py

```
@@ -56,58 +62,189 @@
-        
+        config = _global_memory.get("config", {})
+        token_limit = get_model_token_limit(config) or DEFAULT_TOKEN_LIMIT
+
         # Use REACT agent for Anthropic Claude models, otherwise use CIAYN
-        if provider == 'anthropic' and 'claude' in model_name:
+        if is_anthropic_claude(config):
             logger.debug("Using create_react_agent to instantiate agent.")
-            return create_react_agent(model, tools, checkpointer=checkpointer)
+            agent_kwargs = build_agent_kwargs(checkpointer, config, token_limit)
+            return create_react_agent(model, tools, **agent_kwargs)
         else:
```

Ruff (0.8.2)

Unnecessary `else` after `return` statement

Remove unnecessary `else`

(RET505)